mobile/android/android-device-hash-service.ts

```diff
@@ -0,0 +1,126 @@
+///<reference path="../../.d.ts"/>
+"use strict";
+
+import * as helpers from "../../helpers";
+import * as path from "path";
+import * as temp from "temp";
+
+export class AndroidDeviceHashService {
+	private static HASH_FILE_NAME = "hashes";
+	private static DEVICE_ROOT_PATH = "/data/local/tmp";
+
+	private _hashFileDevicePath: string = null;
+	private _hashFileLocalPath: string = null;
+	private _tempDir: string = null;
+
+	constructor(private adb: Mobile.IAndroidDebugBridge,
+		private appIdentifier: string,
+		private $fs: IFileSystem,
+		private $mobileHelper: Mobile.IMobileHelper) { }
+
+	public get hashFileDevicePath(): string {
+		if (!this._hashFileDevicePath) {
+			this._hashFileDevicePath = this.$mobileHelper.buildDevicePath(AndroidDeviceHashService.DEVICE_ROOT_PATH, this.appIdentifier, AndroidDeviceHashService.HASH_FILE_NAME);
+		}
+
+		return this._hashFileDevicePath;
+	}
+
+	public doesShasumFileExistsOnDevice(): IFuture<boolean> {
+		return ((): boolean => {
+			let lsResult = this.adb.executeShellCommand(["ls", this.hashFileDevicePath]).wait();
```

Comment:
no, no, the idea is to list one file only.
example output of calling ls for existing file is:
```
~ $ ls testInq.js 
testInq.js
```
Output of trying ls of not existing file is:
```
~ $ ls notExisting
ls: cannot access notExisting: No such file or directory
```
The idea is that if the file exists, the output (after trim) will be the same as the file path.